File: python/easy/minimum_moves_to_equal_array_elements.py
class Solution(object):
    def minMoves(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0 or len(nums) == 1:
            return 0
        min_val = sys.maxint
        for num in nums:
            min_val = min(min_val, num)
        ret = 0
        for num in nums:
            ret += num-min_val
        return ret

File: cpp/easy/minimum_moves_to_equal_array_elements.cpp
class Solution {
public:
    int minMoves(vector<int>& nums) {
        if(nums.size() == 0 || nums.size() == 1)
            return 0;
        int mn = INT_MAX, res = 0;
        for (int num : nums) mn = min(mn, num);
        for (int num : nums) res += num - mn;
        return res;
    }
};

Conclusion:

the problem is equal as: find the minimum value in the array, caculate the distance from each element to the minimum value.
